Vagrantfile Documentation

smk4664 opened the pull request:
Adding WIP label as I am sure there will be changes after discussion.

Commit:
Update .gitignore for Jetbrains and VSCode.

Commit:
Add documentation for using a vagrantfile with Self-Medicate.

Commit:
Seperate out minikube instructions to its own page.

Commit:
Seperate Self-Medicate to it's own page.

Commit:
Add new pages to the hacking index.

Commit:
Remove sections that were copied into their own pages.

Commit:
Updating the commands to show the difference between vagrant ssh and minikube ssh.

Mierdin commented:
Dude. Your commit messages are 💯

smk4664 commented:
This is ready to review.  I split the documentation for Self-medicate between running locally vs running through the Vagrantfile.

Mierdin commented on docs/hacking/curriculum.rst:
```diff
@@ -100,129 +49,6 @@ and enter this repository::
-    If your issue isn't covered below, please `open an issue on the
-    selfmedicate repository <https://github.com/nre-learning/antidote-selfmedicate/issues/new>`_.
-
 Cannot connect to the Web Front-End
```
This is fine for now but I'll be following up with https://github.com/nre-learning/antidote-selfmedicate/pull/28 soon which will provide over-arching troubleshooting steps. I'll open a corresponding docs PR to cover that.

smk4664 replied:
I am leaving this in as you noted it is fine. Once the PR is in for the troubleshooting steps, this page can be updated.

Mierdin replied:
:+1:

Mierdin commented on docs/hacking/index.rst:
```diff
@@ -8,5 +8,8 @@ This section is meant for people interested in developing on Antidote.
 
    curriculum.rst
    platform.rst
+   selfmedicate/selfmedicate.rst
```
Either as part of https://github.com/nre-learning/antidote-selfmedicate/pull/28 or very soon after I'm going to just remove the old selfmedicate and make the vagrant approach the only version. It works well and it's time to make the switch.

So to avoid confusion, let's just keep a single method of running selfmedicate in these docs - the vagrant way. I think the "home page" for this directory could be your `selfmedicate.rst` page which covers the selfmedicate script. There should be an introductory section on this page that covers running the script directly (baremetal) or using the vagrant approach (with an internal link to that page).

smk4664 replied:
The changes you suggested has been made.  Please let me know if these changes work.

Mierdin commented on docs/hacking/selfmedicate/selfmedicate.rst:
```diff
@@ -0,0 +1,124 @@
+About Self-Medicate
+===================
+
```
Here's where I am thinking we could have an introductory section that points to the vagrant page as a "primary" way of running selfmedicate (which takes care a lot of the commands below as well), but also offer an explanation that if you're running linux, you can run the script directory and follow the instructions below.

smk4664 replied:
These changes have been made.

Commit:
Updated documentation to show vagrantfile as the preferred method.

Also removed running Minikube locally, now noted as running self-medicate baremetal.
Rearranged the files at Matt's request.

smk4664 reviewed (commented):
@Mierdin Please see the changes and let me know what you think.

Mierdin reviewed (approved):
Thanks. I'll merge this and then when I get around to opening a PR to document the changes in https://github.com/nre-learning/antidote-selfmedicate/pull/28 I'll request your review so we can make sure all these "threads" get merged properly
